modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -1564,18 +1567,17 @@ def cat_codes(self):
         assert len(self.columns) == 1
         assert self._dtypes[-1] == "category"
 
-        col = self.columns[-1]
         exprs = self._index_exprs()
-        col_expr = self.ref(col)
+        col_expr = self.ref(self.columns[-1])
         code_expr = OpExpr("KEY_FOR_STRING", [col_expr], get_dtype("int32"))
         null_val = LiteralExpr(np.int32(-1))
-        exprs[col] = build_if_then_else(
+        exprs[None] = build_if_then_else(
             col_expr.is_null(), null_val, code_expr, get_dtype("int32")
         )
 
         return self.__constructor__(
-            columns=self.columns,
-            dtypes=self._dtypes,
+            columns=Index([None]),
+            dtypes=pd.Series(self._dtypes[0], index=Index([None])),
```

Comment:
why?

Reply:
For compatibility with Pandas. Otherwise, the test test_cat_codes_issue5650 fails with:
```
E           Attribute "name" are different
E           [left]:  name
E           [right]: None
```